File: Regex/Applications/Alien Username.py
import re

username_pattern = r"^[_|.]\d+[a-zA-Z]*_?$"

n = int(input())

for _ in range(n):
    row_input = input()
    if re.search(username_pattern, row_input):
        print("VALID")
    else:
        print("INVALID")

File: Regex/Applications/Alien Username.cpp
#include <cmath>
#include <cstdio>
#include <vector>
#include <iostream>
#include <algorithm>
#include <regex>

using namespace std;


int main() {
    /* Enter your code here. Read input from STDIN. Print output to STDOUT */
    regex username_pattern {"^[_|.]\\d+[a-zA-Z]*_?$"};

    int n;
    cin >> n;
    for (int i = 0; i < n; i++) {
        string row_input;
        cin >> row_input;
        if (regex_search(row_input, username_pattern))
            cout << "VALID" << endl;
        else
            cout << "INVALID" << endl;
    }
    
    return 0;
}
